### ISSUE TITLE
I don't like flathub-json-eol-rebase-misses-new-id

### ISSUE THREAD

**Issue by wjt** [flathub/flatpak-builder-lint#126]
On https://github.com/flathub/org.gnome.dfeet/pull/19 I tried to add the following `flathub.json` to dfeet:

```json
{
    "end-of-life": "This application is no longer maintained and is superseded by D-Spy.",
    "end-of-life-rebase": "org.gnome.dspy"
}
```

The linter rejected this because `flathub-json-eol-rebase-misses-new-id`. The check is:

https://github.com/flathub/flatpak-builder-lint/blob/fc76448d5539fb6a42f1a0b6f4a8982beef17c9e/flatpak_builder_lint/checks/flathub_json.py#L36-L38

So the `end-of-life` message has to include the literal app ID that is specified as `end-of-life-rebase`.

I don't really see why this is necessary. The app ID is present in a machine-readable form already so it's redundant. Is it not better to have the human-readable name of the app in the message?

**Comment by wjt** [comment]
Having merged that PR, I decided to check what different front-ends display in different situations:

## flatpak update (with Flatpak 1.12.8)

```
wjt@camille:~$ flatpak update --no-deploy org.gnome.dfeet
Looking for updates…
Info: org.gnome.dfeet//stable is end-of-life, in favor of app/org.gnome.dspy/x86_64/stable
Replace it with app/org.gnome.dspy/x86_64/stable? [Y/n]: 
```

So it doesn't show the message, only the target ref.

## flatpak install

With `dfeet` uninstalled, then trying to install it:

```
wjt@camille:~$ flatpak install --system flathub org.gnome.dfeet
Looking for matches…
Info: org.gnome.dfeet//stable is end-of-life, in favor of app/org.gnome.dspy/x86_64/stable
Replace it with app/org.gnome.dspy/x86_64/stable? [Y/n]: n
```

## flatpak info

```
       End-of-life: This application is no longer maintained and is superseded by D-Spy (org.gnome.dspy)
End-of-life-rebase: app/org.gnome.dspy/x86_64/stable
```

Displays both.

## flatpak list --columns=application,options

```
org.gnome.dfeet	system,current,eol=This application is no longer maintained and is superseded by D-Spy (org.gnome.dspy),eol-rebase=app/org.gnome.dspy/x86_64/stable
```

Displays both.

## GNOME Software 43

![Screenshot from 2023-03-30 21-03-07](https://user-images.githubusercontent.com/86760/228951519-593112c3-e2bc-4e17-b8bd-736af74b0293.png)

Only uses information derived from the machine-readable app ID.

-----

So in summary, I can't find anywhere that displays the `end-of-life` message without the app ID from the `end-of-life-rebase` key.

**Comment by bbhtt** [comment]
According to barth, they have a vague memory of older flatpak versions, not proposing to rebase and as a result users not knowing which app to switch to. Hence, this error was added.

I didn't find anything like a rebase failure. Doing a search on GitHub, eve in the first preliminary implementation of eol and eol-rebase https://github.com/flatpak/flatpak/pull/1566, it shows the id and the message. This was part of flatpak 0.11.4.

I don't know what the state was pre 0.11.4 or if we should care about versions that old.
